src/__test__/components/data-exploration/heatmap/HeatmapPlot.test.jsx

```diff
@@ -89,26 +84,41 @@ const loadAndRenderDefaultHeatmap = async () => {
   });
 };
 
-const stalledResponse = () => new Promise(() => { });
+const mockAPIResponses = generateDefaultMockAPIResponses(experimentId, fake.PROJECT_ID);
+
+const stalledResponse = () => delayedResponse({ body: 'Not found', status: 404 }, 4000);
 const errorResponse = () => Promise.reject(new Error('Some error idk'));
 
+let storeState = null;
+
 describe('HeatmapPlot', () => {
+  beforeAll(async () => {
+    await preloadAll();
+  });
+
   beforeEach(async () => {
-    enableFetchMocks();
+    jest.clearAllMocks();
+
     fetchMock.resetMocks();
     fetchMock.doMock();
 
-    seekFromAPI.mockClear();
+    fetchMock.mockIf(/.*/, mockAPI(mockAPIResponses));
+
+    seekFromS3
+      .mockReset()
+      .mockImplementationOnce(() => null)
+      .mockImplementationOnce((mockEtag) => mockWorkerResponses[mockEtag]());
```

Comment:
You have the mock s3 in the beforeEach, but then you mock them again in individual tests -- looks like it's better to remove the mock for s3 from the beforeEach in this case

Reply:
unless this mock is actually used within the beforeEach function for the `loadProcessingSettings` and the `loadBackendStatus` actions?

Reply:
No it's not. I'm also putting it here so that this becomes the default behaviour. If the test requires custom behaviour, then the test can implement its own implementation of `mockSeekFromS3`. If we remove it from here, we'll have to add the default implementation for every test, whether the test requires a custom implementation or not, and it would make it difficult to change the base behaviour if we need to.

Reply:
I noticed that the base behaviour is used only in 2 places. The majority of the tests override the behaviour

Reply:
The base case is used in 5 tests, while 4 tests uses custom implementation.

Reply:
yes, you are right actually, somehow I got confused. Cool, let's leave it in the beforeEach